lib/sprockets/utils.rb

```diff
@@ -107,10 +107,8 @@ def concat_javascript_sources(buf, source)
         # This will lead to O(N^2) performance in string_end_with_semicolon?, so we should use 32 bit encoding to make sure indexing stays O(1)
         source = source.encode(Encoding::UTF_32LE) unless source.ascii_only?
 
-        if !string_end_with_semicolon?(source)
-          buf << ";\n"
-        elsif source[source.size - 1].ord != 0x0A
-          buf << "\n"
+        unless string_end_with_semicolon?(source)
+          buf << ";"
         end
```

Comment:
hmm - that was a simple fix.

I was under the impression (which may have been wrongly made) that we may have some sources that do not end in a newline requiring us to need to append one.

I did not realize this line `elsif source[source.size - 1].ord != 0x0A` becomes no longer needed although based on the tests, that feels correct.